akvo/rsr/views/py_reports/project_results_indicators_excel_report.py

```diff
@@ -0,0 +1,183 @@
+                font=Font(size=12, color=Color(255, 255, 255)),
+                alignment=Alignment(wrap_text=True),
+                fill=Fill(background=Color(89, 89, 89)))
+            ws.range('A' + str(row), 'B' + str(row)).merge()
+            ws.set_cell_style(row, 1, result_header2_style)
+            ws.set_cell_value(row, 1, result.title)
+            ws.range('C' + str(row), ('N' if has_disaggregation else 'L') + str(row)).merge()
+            ws.set_cell_style(row, 3, result_header2_style)
+            ws.set_cell_value(row, 3, result.description)
+            row += 1
+
+            # r7
+            ws.set_row_style(row, Style(size=36))
+            row7_style = Style(
+                font=Font(bold=True, size=12),
+                fill=Fill(background=Color(211, 211, 211)))
+            for i in range(1, max_column + 1):
+                ws.set_cell_style(row, i, row7_style)
+            ws.set_cell_value(row, 1, 'Indicator title')
+            ws.set_cell_value(row, 2, 'Indicator description')
+            ws.set_cell_value(row, 3, 'Baseline year')
+            ws.set_cell_value(row, 4, 'Baseline value')
+            ws.set_cell_value(row, 5, 'Baseline comment')
+            ws.set_cell_value(row, 6, 'Period start')
+            ws.set_cell_value(row, 7, 'Period end')
+            ws.set_cell_value(row, 8, 'Target value')
+            ws.set_cell_value(row, 9, 'Target comment')
+            ws.set_cell_value(row, 10, 'Actual value')
+            ws.set_cell_value(row, 11, 'Actual comment')
+            if has_disaggregation:
+                ws.set_cell_value(row, 12, 'Disaggregation label')
+                ws.set_cell_value(row, 13, 'Disaggregation value')
+            ws.set_cell_value(row, max_column, 'Aggregation status')
+            row += 1
+
+            ws.set_cell_value(row, max_column, 'Yes' if result.aggregation_status else 'No')
+            for indicator in result.indicators.all():
+                # r8
+                ws.set_cell_style(row, 1, Style(alignment=Alignment(wrap_text=True)))
+                ws.set_cell_value(row, 1, indicator.title)
+                ws.set_cell_style(row, 2, Style(alignment=Alignment(wrap_text=True)))
+                ws.set_cell_value(row, 2, indicator.description)
+                ws.set_cell_style(row, 3, Style(alignment=Alignment(horizontal='right')))
+                ws.set_cell_value(row, 3, indicator.baseline_year)
+                ws.set_cell_style(row, 4, Style(alignment=Alignment(horizontal='right')))
+                ws.set_cell_value(row, 4, indicator.baseline_value)
+                ws.set_cell_style(row, 5, Style(alignment=Alignment(wrap_text=True)))
+                ws.set_cell_value(row, 5, indicator.baseline_comment)
+
+                for period in indicator.periods.all():
+                    ws.set_cell_value(
+                        row, 6,
+                        utils.get_period_start(period, in_eutf_hierarchy).strftime('%Y-%m-%d'))
```

Comment:
There can be projects where the period dates have not been set correctly, and this would cause a server error. May be handle `start_date` or `end_date` being `None`?

Reply:
Good catch! Thanks @punchagan